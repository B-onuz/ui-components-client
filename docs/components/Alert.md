# Alert (Component)

Componente de para avisos do sistema.

## Como utilizar

```js
import Alert from 'ui-components/lib/components/Alert'
```

```jsx
<Alert m={10} color={'warning'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
<Alert m={10} color={'success'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
<Alert m={10} color={'info'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
<Alert m={10} color={'danger'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
```

## Variações

### Variant

```jsx
<Alert color={'primary'} />
```

Escala no tema: `theme.colors`

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
