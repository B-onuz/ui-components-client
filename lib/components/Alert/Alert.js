"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _color = _interopRequireDefault(require("color"));

var _styledSystem = require("styled-system");

var _variants = require("./variants");

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Alert/Alert.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  margin: 0;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  margin: 0 0 8px;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-right: 16px;\n  font-size: ", ";\n  opacity: 1;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  width: ", ";\n  height: ", ";\n  margin-right: 16px;\n  background: ", ";\n  background-color: ", ";\n  background-size: cover;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  background: #fff;\n  border-radius: 4px;\n  ", ";\n  @keyframes slide_bottom {\n    from {\n      transform: translateY(100%);\n      opacity: 0.5;\n    }\n    to {\n      transform: translateY(0);\n      opacity: 1;\n    }\n  }\n  @keyframes slide_top {\n    from {\n      transform: translateY(-100%);\n      opacity: 0.5;\n    }\n    to {\n      transform: translateY(0);\n      opacity: 1;\n    }\n  }\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  border: ", ";\n  border-radius: 4px;\n  padding: ", ";\n  display: flex;\n  justify-content: flex-start;\n  align-items: center;\n  box-shadow: ", ";\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledAlert = _styledComponents["default"].div(_templateObject(), (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout), function (_ref) {
  var noBorder = _ref.noBorder;
  return !!noBorder ? 'none' : '1px solid';
}, function (_ref2) {
  var shortAlert = _ref2.shortAlert;
  return !!shortAlert ? 0 : '12px 24px';
}, function (_ref3) {
  var noShadow = _ref3.noShadow;
  return noShadow ? 'none !important' : '';
}, _variants.color);

var Wrapper = _styledComponents["default"].div(_templateObject2(), function (props) {
  return props.fullWidth && 'width: 100%' || '';
}, function (_ref4) {
  var animate = _ref4.animate;
  return "\n    animation-name: slide_" + animate + ";\n    animation-duration: .3s;\n  ";
});

var StyledIcon = _styledComponents["default"].div(_templateObject3(), function (_ref5) {
  var imgSize = _ref5.imgSize;
  return imgSize;
}, function (_ref6) {
  var imgSize = _ref6.imgSize;
  return imgSize;
}, function (_ref7) {
  var imgBackground = _ref7.imgBackground;
  return "url(" + imgBackground + ") no-repeat center center";
}, function (_ref8) {
  var bgColor = _ref8.bgColor;
  return bgColor;
});

var StyledFaIcon = _styledComponents["default"].div(_templateObject4(), function (_ref9) {
  var iconSize = _ref9.iconSize;
  return iconSize ? iconSize : '32px';
});

var StyledTitle = _styledComponents["default"].h3(_templateObject5());

var StyledMessage = _styledComponents["default"].p(_templateObject6());

var Alert = function Alert(_ref10) {
  var children = _ref10.children,
      animate = _ref10.animate,
      title = _ref10.title,
      message = _ref10.message,
      icon = _ref10.icon,
      iconSize = _ref10.iconSize,
      customIcon = _ref10.customIcon,
      imgSize = _ref10.imgSize,
      imgBackground = _ref10.imgBackground,
      bgColor = _ref10.bgColor,
      noShadow = _ref10.noShadow,
      _ref10$type = _ref10.type,
      type = _ref10$type === void 0 ? {} : _ref10$type,
      shortAlert = _ref10.shortAlert,
      noBorder = _ref10.noBorder,
      fullWidth = _ref10.fullWidth,
      rest = _objectWithoutPropertiesLoose(_ref10, ["children", "animate", "title", "message", "icon", "iconSize", "customIcon", "imgSize", "imgBackground", "bgColor", "noShadow", "type", "shortAlert", "noBorder", "fullWidth"]);

  return /*#__PURE__*/_react["default"].createElement(Wrapper, {
    fullWidth: fullWidth,
    animate: animate,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 92,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(StyledAlert, _extends({}, rest, {
    noShadow: noShadow,
    shortAlert: shortAlert,
    noBorder: noBorder,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 93,
      columnNumber: 7
    }
  }), icon && /*#__PURE__*/_react["default"].createElement(StyledFaIcon, {
    iconSize: iconSize,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 95,
      columnNumber: 11
    }
  }, /*#__PURE__*/_react["default"].createElement(_reactFontawesome.FontAwesomeIcon, {
    icon: icon,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 96,
      columnNumber: 13
    }
  })), imgBackground && /*#__PURE__*/_react["default"].createElement(StyledIcon, {
    imgBackground: imgBackground,
    imgSize: imgSize,
    bgColor: bgColor,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 99,
      columnNumber: 27
    }
  }), /*#__PURE__*/_react["default"].createElement("div", {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 100,
      columnNumber: 9
    }
  }, !!title && /*#__PURE__*/_react["default"].createElement(StyledTitle, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 101,
      columnNumber: 23
    }
  }, title), /*#__PURE__*/_react["default"].createElement(StyledMessage, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 102,
      columnNumber: 11
    }
  }, message)), children));
};

Alert.defaultProps = {
  shortAlert: false
};
var _default = Alert;
exports["default"] = _default;
module.exports = exports.default;