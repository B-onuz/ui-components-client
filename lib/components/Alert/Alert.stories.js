'use strict'

exports.__esModule = true
exports.FixedAlertCustomAlert = exports.FixedAlert = exports.AlertLongText = exports.AlertColors = exports.AlertExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Alert = _interopRequireDefault(require('./Alert'))

var _reactFontawesome = require('@fortawesome/react-fontawesome')

var _freeSolidSvgIcons = require('@fortawesome/free-solid-svg-icons')

var _reflexbox = require('reflexbox')

var _Typography = _interopRequireDefault(require('../Typography'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Alert/Alert.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Alert',
  component: _Alert['default'],
}
exports['default'] = _default

var AlertExample = function AlertExample() {
  return /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
    m: 10,
    color: 'success',
    title: 'Exemplo de t\xEDtulo',
    message: 'Este texto \xE9 exemplo de uma message',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 35,
    },
  })
}

exports.AlertExample = AlertExample

var AlertColors = function AlertColors() {
  return /*#__PURE__*/ _react['default'].createElement(
    'div',
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 16,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      m: 10,
      color: 'warning',
      title: 'Exemplo de t\xEDtulo',
      message: 'Este texto \xE9 exemplo de uma message',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 17,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      m: 10,
      color: 'success',
      title: 'Exemplo de t\xEDtulo',
      message: 'Este texto \xE9 exemplo de uma message',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 18,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      m: 10,
      color: 'info',
      title: 'Exemplo de t\xEDtulo',
      message: 'Este texto \xE9 exemplo de uma message',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 19,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      m: 10,
      color: 'danger',
      title: 'Exemplo de t\xEDtulo',
      message: 'Este texto \xE9 exemplo de uma message',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 20,
        columnNumber: 5,
      },
    })
  )
}

exports.AlertColors = AlertColors

var AlertLongText = function AlertLongText() {
  return /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
    m: 10,
    color: 'success',
    title: 'Exemplo de t\xEDtulo',
    message:
      'Laboris consequat consequat voluptate qui nisi nostrud dolor. Velit adipisicing id in elit enim pariatur enim aliquip exercitation exercitation duis in qui. Sint exercitation aute enim deserunt. Non labore aute mollit cillum Lorem incididunt pariatur nostrud ut. Quis exercitation et culpa est. Consectetur consectetur fugiat nostrud in nostrud. Lorem consectetur culpa cillum nostrud adipisicing anim veniam id Lorem Lorem elit ad.\n\nNostrud fugiat exercitation tempor sunt amet nulla. Qui ipsum laborum ut deserunt esse sint pariatur labore ipsum pariatur. Culpa ipsum et cupidatat sit esse aliquip sunt. Sunt adipisicing veniam adipisicing nisi dolor dolor labore id officia Lorem laborum.\n\nUllamco eiusmod ipsum duis ad. Anim commodo sint voluptate fugiat ut ipsum nostrud mollit laboris consectetur laboris sit consectetur. Esse do velit pariatur cillum reprehenderit et elit ullamco cillum excepteur proident laborum pariatur.',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 3,
    },
  })
}

exports.AlertLongText = AlertLongText

var FixedAlert = function FixedAlert() {
  return /*#__PURE__*/ _react['default'].createElement(
    'div',
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 38,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      m: 10,
      color: 'success',
      title: 'Exemplo de t\xEDtulo',
      message: 'Este texto \xE9 exemplo de uma message',
      imgBackground: 'https://i.pravatar.cc/300',
      imgSize: '50px',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 39,
        columnNumber: 5,
      },
    }),
    /*#__PURE__*/ _react['default'].createElement(_Alert['default'], {
      pl: 1,
      m: 10,
      color: 'black',
      noBorder: true,
      title: 'Exemplo de t\xEDtulo',
      noShadow: true,
      message: 'Este texto \xE9 exemplo de uma message',
      icon: _freeSolidSvgIcons.faHandshake,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 47,
        columnNumber: 5,
      },
    })
  )
}

exports.FixedAlert = FixedAlert

var FixedAlertCustomAlert = function FixedAlertCustomAlert() {
  return /*#__PURE__*/ _react['default'].createElement(
    _Alert['default'],
    {
      m: 10,
      color: 'black',
      noShadow: true,
      shortAlert: true,
      fullWidth: true,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 61,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Flex,
      {
        p: 2,
        justifyContent: 'space-between',
        style: {
          width: '100%',
        },
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 62,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _reflexbox.Flex,
        {
          alignItems: 'center',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 63,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(_reactFontawesome.FontAwesomeIcon, {
          icon: _freeSolidSvgIcons.faCalendarAlt,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 64,
            columnNumber: 9,
          },
        }),
        /*#__PURE__*/ _react['default'].createElement(
          _Typography['default'],
          {
            m: 0,
            ml: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 65,
              columnNumber: 9,
            },
          },
          '15 dias \xFAteis'
        )
      ),
      /*#__PURE__*/ _react['default'].createElement(
        _reflexbox.Flex,
        {
          alignItems: 'center',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 69,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(_reactFontawesome.FontAwesomeIcon, {
          icon: _freeSolidSvgIcons.faDollarSign,
          color: 'grey',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 70,
            columnNumber: 9,
          },
        }),
        /*#__PURE__*/ _react['default'].createElement(
          _Typography['default'],
          {
            m: 0,
            ml: 2,
            color: 'primary',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 71,
              columnNumber: 9,
            },
          },
          '250 BNZs'
        )
      )
    )
  )
}

exports.FixedAlertCustomAlert = FixedAlertCustomAlert
