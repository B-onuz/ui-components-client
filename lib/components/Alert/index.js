"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Alert = _interopRequireDefault(require("./Alert"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Alert["default"];
exports["default"] = _default;
module.exports = exports.default;