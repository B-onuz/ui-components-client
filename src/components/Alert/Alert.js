import React from 'react'
import styled from 'styled-components'
import Color from 'color'
import { space, compose, layout } from 'styled-system'
import { color } from './variants'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const StyledAlert = styled.div`
  ${compose(space, layout)}
  border: ${({ noBorder }) => (!!noBorder ? 'none' : '1px solid')};
  border-radius: 4px;
  padding: ${({ shortAlert }) => (!!shortAlert ? 0 : '12px 24px')};
  display: flex;
  justify-content: flex-start;
  align-items: center;
  box-shadow: ${({ noShadow }) => (noShadow ? 'none !important' : '')};
  ${color};
`

const Wrapper = styled.div`
  background: #fff;
  border-radius: 4px;
  ${(props) => (props.fullWidth && 'width: 100%') || ''};
  @keyframes slide_bottom {
    from {
      transform: translateY(100%);
      opacity: 0.5;
    }
    to {
      transform: translateY(0);
      opacity: 1;
    }
  }
  @keyframes slide_top {
    from {
      transform: translateY(-100%);
      opacity: 0.5;
    }
    to {
      transform: translateY(0);
      opacity: 1;
    }
  }
  ${({ animate }) => `
    animation-name: slide_${animate};
    animation-duration: .3s;
  `}
`

const StyledIcon = styled.div`
  width: ${({ imgSize }) => imgSize};
  height: ${({ imgSize }) => imgSize};
  margin-right: 16px;
  background: ${({ imgBackground }) => `url(${imgBackground}) no-repeat center center`};
  background-color: ${({ bgColor }) => bgColor};
  background-size: cover;
`

const StyledFaIcon = styled.div`
  margin-right: 16px;
  font-size: ${({ iconSize }) => (iconSize ? iconSize : '32px')};
  opacity: 1;
`

const StyledTitle = styled.h3`
  margin: 0 0 8px;
`

const StyledMessage = styled.p`
  margin: 0;
`

const Alert = ({
  children,
  animate,
  title,
  message,
  icon,
  iconSize,
  customIcon,
  imgSize,
  imgBackground,
  bgColor,
  noShadow,
  type = {},
  shortAlert,
  noBorder,
  fullWidth,
  ...rest
}) => {
  return (
    <Wrapper fullWidth={fullWidth} animate={animate}>
      <StyledAlert {...rest} noShadow={noShadow} shortAlert={shortAlert} noBorder={noBorder}>
        {icon && (
          <StyledFaIcon iconSize={iconSize}>
            <FontAwesomeIcon icon={icon} />
          </StyledFaIcon>
        )}
        {imgBackground && <StyledIcon imgBackground={imgBackground} imgSize={imgSize} bgColor={bgColor} />}
        <div>
          {!!title && <StyledTitle>{title}</StyledTitle>}
          <StyledMessage>{message}</StyledMessage>
        </div>
        {children}
      </StyledAlert>
    </Wrapper>
  )
}

Alert.defaultProps = {
  shortAlert: false,
}

export default Alert
