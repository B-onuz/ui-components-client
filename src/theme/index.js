const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
