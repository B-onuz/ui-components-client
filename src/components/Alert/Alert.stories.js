import React from 'react'
import Alert from './Alert'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faArchive, faHandshake, faCalendarAlt, faDollarSign } from '@fortawesome/free-solid-svg-icons'
import { Box, Flex } from 'reflexbox'
import Typography from '../Typography'

export default {
  title: 'Alert',
  component: Alert,
}

export const AlertExample = () => <Alert m={10} color={'success'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />

export const AlertColors = () => (
  <div>
    <Alert m={10} color={'warning'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
    <Alert m={10} color={'success'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
    <Alert m={10} color={'info'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
    <Alert m={10} color={'danger'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
  </div>
)

export const AlertLongText = () => (
  <Alert
    m={10}
    color={'success'}
    title={`Exemplo de título`}
    message={`Laboris consequat consequat voluptate qui nisi nostrud dolor. Velit adipisicing id in elit enim pariatur enim aliquip exercitation exercitation duis in qui. Sint exercitation aute enim deserunt. Non labore aute mollit cillum Lorem incididunt pariatur nostrud ut. Quis exercitation et culpa est. Consectetur consectetur fugiat nostrud in nostrud. Lorem consectetur culpa cillum nostrud adipisicing anim veniam id Lorem Lorem elit ad.

Nostrud fugiat exercitation tempor sunt amet nulla. Qui ipsum laborum ut deserunt esse sint pariatur labore ipsum pariatur. Culpa ipsum et cupidatat sit esse aliquip sunt. Sunt adipisicing veniam adipisicing nisi dolor dolor labore id officia Lorem laborum.

Ullamco eiusmod ipsum duis ad. Anim commodo sint voluptate fugiat ut ipsum nostrud mollit laboris consectetur laboris sit consectetur. Esse do velit pariatur cillum reprehenderit et elit ullamco cillum excepteur proident laborum pariatur.`}
  />
)

export const FixedAlert = () => (
  <div>
    <Alert
      m={10}
      color={'success'}
      title={`Exemplo de título`}
      message={`Este texto é exemplo de uma message`}
      imgBackground={'https://i.pravatar.cc/300'}
      imgSize={'50px'}
    />
    <Alert
      pl={1}
      m={10}
      color={'black'}
      noBorder
      title={`Exemplo de título`}
      noShadow
      message={`Este texto é exemplo de uma message`}
      icon={faHandshake}
    />
  </div>
)

export const FixedAlertCustomAlert = () => (
  <Alert m={10} color={'black'} noShadow shortAlert fullWidth>
    <Flex p={2} justifyContent="space-between" style={{ width: '100%' }}>
      <Flex alignItems="center">
        <FontAwesomeIcon icon={faCalendarAlt} />
        <Typography m={0} ml={2}>
          15 dias úteis
        </Typography>
      </Flex>
      <Flex alignItems="center">
        <FontAwesomeIcon icon={faDollarSign} color={'grey'} />
        <Typography m={0} ml={2} color="primary">
          250 BNZs
        </Typography>
      </Flex>
    </Flex>
  </Alert>
)
