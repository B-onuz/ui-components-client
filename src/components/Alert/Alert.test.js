import React from 'react'
import Alert from './Alert.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'

test('Alert', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <>
        <Alert m={10} color={'warning'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
        <Alert m={10} color={'success'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
        <Alert m={10} color={'info'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
        <Alert m={10} color={'danger'} title={`Exemplo de título`} message={`Este texto é exemplo de uma message`} />
      </>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
