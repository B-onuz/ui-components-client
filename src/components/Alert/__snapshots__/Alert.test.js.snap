// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert 1`] = `
Array [
  .c1 {
  margin: 10px;
  border: 1px solid;
  border-radius: 4px;
  padding: 12px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #7A4D05;
  background-color: hsla(36.89999999999998,100%,49.8%,0.09999999999999998);
  box-shadow: 0 0 3px hsl(36.89999999999998,100%,49.8%);
}

.c1 > p {
  color: rgba(122,77,5,0.85);
}

.c0 {
  background: #fff;
  border-radius: 4px;
  -webkit-animation-name: slide_undefined;
  animation-name: slide_undefined;
  -webkit-animation-duration: .3s;
  animation-duration: .3s;
}

.c2 {
  margin: 0 0 8px;
}

.c3 {
  margin: 0;
}

<div
    className="c0"
  >
    <div
      className="c1"
      color="warning"
    >
      <div>
        <h3
          className="c2"
        >
          Exemplo de título
        </h3>
        <p
          className="c3"
        >
          Este texto é exemplo de uma message
        </p>
      </div>
    </div>
  </div>,
  .c1 {
  margin: 10px;
  border: 1px solid;
  border-radius: 4px;
  padding: 12px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1E561F;
  background-color: hsla(121.10000000000002,100%,45.5%,0.09999999999999998);
  box-shadow: 0 0 3px hsl(121.10000000000002,100%,45.5%);
}

.c1 > p {
  color: rgba(30,86,31,0.85);
}

.c0 {
  background: #fff;
  border-radius: 4px;
  -webkit-animation-name: slide_undefined;
  animation-name: slide_undefined;
  -webkit-animation-duration: .3s;
  animation-duration: .3s;
}

.c2 {
  margin: 0 0 8px;
}

.c3 {
  margin: 0;
}

<div
    className="c0"
  >
    <div
      className="c1"
      color="success"
    >
      <div>
        <h3
          className="c2"
        >
          Exemplo de título
        </h3>
        <p
          className="c3"
        >
          Este texto é exemplo de uma message
        </p>
      </div>
    </div>
  </div>,
  .c1 {
  margin: 10px;
  border: 1px solid;
  border-radius: 4px;
  padding: 12px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0E566C;
  background-color: hsla(194,100%,47.8%,0.09999999999999998);
  box-shadow: 0 0 3px hsl(194,100%,47.8%);
}

.c1 > p {
  color: rgba(14,86,108,0.85);
}

.c0 {
  background: #fff;
  border-radius: 4px;
  -webkit-animation-name: slide_undefined;
  animation-name: slide_undefined;
  -webkit-animation-duration: .3s;
  animation-duration: .3s;
}

.c2 {
  margin: 0 0 8px;
}

.c3 {
  margin: 0;
}

<div
    className="c0"
  >
    <div
      className="c1"
      color="info"
    >
      <div>
        <h3
          className="c2"
        >
          Exemplo de título
        </h3>
        <p
          className="c3"
        >
          Este texto é exemplo de uma message
        </p>
      </div>
    </div>
  </div>,
  .c1 {
  margin: 10px;
  border: 1px solid;
  border-radius: 4px;
  padding: 12px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #973937;
  background-color: hsla(1.3000000000000114,100%,80.8%,0.09999999999999998);
  box-shadow: 0 0 3px hsl(1.3000000000000114,100%,80.8%);
}

.c1 > p {
  color: rgba(151,57,55,0.85);
}

.c0 {
  background: #fff;
  border-radius: 4px;
  -webkit-animation-name: slide_undefined;
  animation-name: slide_undefined;
  -webkit-animation-duration: .3s;
  animation-duration: .3s;
}

.c2 {
  margin: 0 0 8px;
}

.c3 {
  margin: 0;
}

<div
    className="c0"
  >
    <div
      className="c1"
      color="danger"
    >
      <div>
        <h3
          className="c2"
        >
          Exemplo de título
        </h3>
        <p
          className="c3"
        >
          Este texto é exemplo de uma message
        </p>
      </div>
    </div>
  </div>,
]
`;
